### PR TITLE
Add renew connection data callback

### PR DESCRIPTION
**Pull request by Holofox**
Hi! Currently it is possible to set connection data only when creating a class instance. My changes allow changing this data on the fly at any time.

### CODE REVIEW

File: lib/src/client.dart [hunk 1]
```diff
@@ -463,8 +472,9 @@ class ClientImpl implements Client {
     if (_token != '') {
       request.token = _token;
     }
-    if (_data != null) {
-      request.data = _data!;
+    final data = await _config.getData?.call() ?? _data;
```
Comment (FZambia) [inline lib/src/client.dart:475]: I think if getData is provided - its returned value should always be used, so it's possible to clean up _data. I.e.:

```
final data = _config.getData != null ? await _config.getData!.call() : _data;
```

File: lib/src/client.dart [hunk 2]
```diff
@@ -50,6 +50,10 @@ abstract class Client {
   ///
   void setToken(String token);
 
+  /// Set allows updating connection data.
+  ///
+  void setData(List<int> data);
```
Comment (FZambia) [inline lib/src/client.dart:55]: Do you need both `getData` callback and `setData` setter? Can we go only with callback only (in this case we will match interface in centrifuge-js - https://github.com/centrifugal/centrifuge-js/blob/9749053dfb669cd559cf0dd50b3f3b7f209a0e02/src/types.ts#L103).

Comment (Holofox) [reply lib/src/client.dart:55]: @FZambia, basically, only a callback is needed, a getter and setter are not needed, I will do it like in the js version.

Comment (Holofox) [reply lib/src/client.dart]: @FZambia, i made corrections, please look again

Comment (FZambia) [reply lib/src/client.dart]: Thanks 👍

**Comment by FZambia** [comment]
Many thanks, released in [v0.14.1](https://github.com/centrifugal/centrifuge-dart/releases/tag/v0.14.1)

**Pull request merged**